made enigma keyboard accessible

kywoodland opened the pull request:
Before, it was impossible to tab into the widget with the keyboard, but now i've added tabindex to the appropriate elements and added functions to show/hide tabindex depending on which "screen" the user is on (so they can't tab hidden elements).

kywoodland committed:
added tabindex, created functions to control tab access

kywoodland committed:
deleted checkKey function and replaced with ng-keyup statement

iturgeon commented on src/controllers/player.coffee:
```diff
@@ -22,6 +22,11 @@ Enigma.controller 'enigmaPlayerCtrl', ['$scope', '$timeout', ($scope, $timeout)
 	$scope.delayedHeaderInit = false
 
+	# variable to check which screen the user is on (true = gameboard)
+	$scope.checkTab = true
```
OMG

iturgeon commented on src/controllers/player.coffee:
```diff
@@ -67,6 +75,15 @@ Enigma.controller 'enigmaPlayerCtrl', ['$scope', '$timeout', ($scope, $timeout)
+
+	# function to find the first unanswered question in list and shift focus to it
+	$scope.findQuestion = ->
+		for item in document.getElementsByClassName('question')
```
IDK, maybe this could be better

iturgeon reviewed (changes requested):
DO IT

clpetersonucf reviewed (changes requested):
This is good so far!! I have a handful of suggestions, just to improve usability.

First off, this is totally playable now. These suggestions are more about improving awareness of the state of the widget at any given time.

- The initial tab into the widget brings right directly to the first item on the first row. It might be handy to have an a hidden element that serves as the first tab index with an `aria-label` that provides instructions, or simply add it to the title of the widget, so users initially know what's going on ("Widget name" followed by "Questions are sorted into categories. Begin by tabbing through and responding to each question within each category", etc.)
- Grid items should probably announce what category they belong to, because right now that's not spoken to the user I don't think. "Animated TV: Question 1" and perhaps the state, if it's been answered ("Correct" or "Incorrect")
- Within individual questions, tabbing through the multiple choice responses works great, but again it might be beneficial to announce which one you've selected. Similarly, once an item has been selected and you tab to "Submit final answer", announcing the new state once you return to the question grid could be handy. ("You've answered 5 of 9 questions", etc.)

`aria-live` is great for this, because you can change the content of a single aria-live element and it immediately is spoken to the user regardless of what they're focused on within the DOM. It's great for state-related feedback that doesn't necessarily belong within an `aria-label` on every element.

kywoodland committed:
pr comment updates; adding aria-labels to things

kywoodland committed:
updated based on corey's comment; fixed other minor issues

kywoodland commented:
Updated the things based on Corey's comment:

1. Added a quick introduction that the screen reader reads when tabbed into the widget, directly after the welcome heading
2. For every question on the game board, the screen reader says the category and the question number, and if it's answered or correct/incorrect
3. If a user selects the question using the screen reader select buttons (spacebar for Voice Over, for example) it announces which was selected. And then reads back how many questions are remaining

Also:

- added live feedback via screen reader, so the user immediately knows if their answer was correct or what the correct answer was (or if they were partially correct)
- fixed the issue of the screen reader repeating the "Questions Remaining" twice
- got rid of some redundancies so the user doesn't have to tab through the same element 3 times to get to the next one, as was the case with each category

clpetersonucf reviewed (commented):
This is already very usable! There's just a few final things that seem a little off, at least when using ChromeVox:

- If I tab into the widget, it focuses the header that says "welcome to Enigma!", but I haven't been able to navigate to the instructions after that. The only way seems to be to go forward with the arrow keys (down or right), then backtrack to focus the instructions. What if we combined the instructions with the "welcome to enigma!" label?

- I'm not noticing the aria-live region ever being spoken, either after choosing an answer, submitting it, or returning to the game board. Does it work with other screen readers?

- This is really minor, but there's no submit option prior to completing the widget - and when the final question is answered, the existing dom is basically replaced by the submit button. If I was unable to see what was happening on the screen, I might be a little confused. We should probably either add an aria-live update to announce what happened "You completed all questions, you scored 9 out of 9 correct, select submit to go to the score screen" or maybe stay on the game board and have a submit option available after the game board element.

kywoodland committed:
made changes based on feedback

clpetersonucf committed:
Fixes for tests, added ng-aria, corrected score circle animations

clpetersonucf committed:
Fixed final failing test

clpetersonucf committed:
tweaked test coverage thresholds

clpetersonucf reviewed (approved):
I think we're good.
